addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3641,6 +3641,15 @@ var SnippetsMenu = Widget.extend({
         }
 
         this.$('.o_snippet_search_filter').toggleClass('d-none', this._currentTab !== this.tabs.BLOCKS);
+        const searchInput = this.el.querySelector('.o_snippet_search_filter_input');
+        // Focus the search input when switching to the blocks tab.
+        if (this._currentTab === this.tabs.BLOCKS) {
+            searchInput.focus();
+        }
+        // Clear the search input when switching to the options tab.
+        if (searchInput.value !== '') {
```

Comment:
OK, we will keep it in mind for forward-port in master

Reply:
Agreed ! I suggest we can have a common method for both the case i.e `_onSnippetSearchResetClick` => `_snippetSearchReset`, i know it would require me to change in xml as well, but having two methods doing the same thing is not a great idea. i will place a comment on this line.